packages/generator/ni_measurement_plugin_sdk_generator/client/__init__.py

```diff
@@ -47,25 +48,256 @@ def _create_file(
+    if len(measurement_service_class) == 0:
+        raise click.ClickException("No registered measurements.")
+
+    for service_class in measurement_service_class:
+        base_service_class = _extract_base_service_class(service_class)
+        module_name = _create_module_name(base_service_class)
+        class_name = _create_class_name(base_service_class)
+        _validate_identifier(module_name, "module")
+        _validate_identifier(class_name, "class")
+
+        _create_client(
+            channel_pool=channel_pool,
+            discovery_client=discovery_client,
+            measurement_service_class=service_class,
+            module_name=module_name,
+            class_name=class_name,
+            directory_out=directory_out_path,
+        )
+
+
+def _create_clients_interactively() -> None:
+    print("Creating the Python Measurement Plug-In Client in interactive mode...")
+    channel_pool = GrpcChannelPool()
+    discovery_client = DiscoveryClient(grpc_channel_pool=channel_pool)
+    directory_out_path = _resolve_output_directory()
+
+    while True:
+        measurement_service_class, measurement_display_names = get_all_registered_measurement_info(
+            discovery_client
+        )
+        if len(measurement_service_class) == 0:
+            raise click.ClickException("No registered measurements.")
+
+        print("\nList of registered measurements:")
+        for index, display_name in enumerate(measurement_display_names, start=1):
+            print(f"{index}. {display_name}")
+
+        selection = click.prompt(
+            "\nSelect a measurement to generate a client (x to exit)",
+            type=str,
+        )
+        if selection.lower() == "x":
+            break
+        service_class = _get_selected_measurement_service_class(
+            int(selection), measurement_service_class
+        )
+
+        base_service_class = _extract_base_service_class(service_class)
+        default_module_name = _create_module_name(base_service_class)
+        module_name = click.prompt(
+            "Enter a name for the Python client module (or) press enter to choose the default name",
+            type=str,
+            default=default_module_name,
+        )
+        _validate_identifier(module_name, "module")
+        default_class_name = _create_class_name(base_service_class)
+        class_name = click.prompt(
+            "Enter a name for the Python client class (or) press enter to choose the default name",
+            type=str,
+            default=default_class_name,
+        )
+        _validate_identifier(class_name, "class")
+
+        _create_client(
+            channel_pool=channel_pool,
+            discovery_client=discovery_client,
+            measurement_service_class=service_class,
+            module_name=module_name,
+            class_name=class_name,
+            directory_out=directory_out_path,
+        )
+
+
+def _create_clients(
+    measurement_service_class: List[str],
```

Comment:
Probably I overlooked this in several instances but this name doesn't emphasize the fact that it holds a list of service class.
```suggestion
    measurement_service_classes: List[str],
```